articles/nodejs-structural-comparisons/index.md

```diff
@@ -0,0 +1,562 @@
+
+#### Table of Contents
+* [Introduction](#introduction)
+* [Prerequisites](#prerequisites)
+* [Modules and Libraries](#modules-and-libraries)
+* [Variables with Value](#variables-with-value)
+* [Graphing](#graphing)
+* [Graphical User Interface Applications](#graphical-user-interface-applications)
+
+#### Prerequisites
+
+* Windows device.
+* Previous knowledge of installing modules and libraries.
+* R-Programming.
+* Kali Linux (to compile C and C++ codes).
+* NetBeans (to build in Java).
+* Node.js application (to write in Node.js).
+* Plotly account (to graph).
+* Some coding knowledge or experience.
+
+Once most of the prerequisites are met, let's begin.  
+
+#### Modules and Libraries
+
+##### Node.js
+
+Node.js computer language is written in a unique structure. One characteristic to consider is the steps to create or add modules into Node.js. As each coding language includes its sequenced procedure, Node.js also involves flexible options.  
+
+We can find Node.js modules on open-source websites. In Node.js, folders and JavaScript files are known to hold modules. Modules inside the `node_module` folder can be used in the Node.js application.  
+
+A branched-out path directory looks similar to the picture below.  
+
+[directory](/engineering-education/nodejs-structural-comparisons/path.jpg)  
+_Screen capture_  
+
+Remember to use the `npm rebuild` command.  
+
+```JavaScript
+global.puny = require('punycode')
+```  
+
+Functions for the module should appear at the bottom of the Node.js application window.  
+
+The choice to include dependencies is usually not available. This module can be used with the codes shown below.  
+
+```JavaScript
+global.char = 'sectionio';
+global.changeto = puny.decode(char);
+```  
+
+**Output:**  
+```JavaScript
+'ઽ઺ૃરીૄ૆'
+```  
+
+The code snippet below shows how to encode the above output back to the original text string variable.  
+
+```JavaScript
+puny.encode(changeto);
+```  
+
+**Output:**  
+```JavaScript
+'sectionio'
+```  
+
+Shown below is how the code is displayed in a Node.js window.  
+
+[punycode code snippet](/engineering-education/nodejs-structural-comparisons/punycodenodejs.jpg)  
+_Screen capture_  
+
+As a side note, installing external Node.js modules can be found in the documentation.  
+
+##### R-Programming
+
+More information can be found in [CRAN documentation](https://cran.r-project.org). Further options can include `dependencies =` and `source = `.  
+
+Similar to Node.js, R-Programming also uses browser windows to display. Usually, in R-Programming, RStudio can display visuals. However, Graphic User Interface (GUI) applications are sent to an internet browser window.  
+
+##### C/C++
+
+The following is the most frequently used procedure to install a C module. An advanced statistical module will be installed. More information on how to install the TensorFlow library can be found [here](https://www.tensorflow.org/install/lang_c).  
```

Comment:
TensorFlow is not a statistical library. Rewrite the sentence accurately.

Reply:
What I was attempting to mention was even though the simple view of TensorFlow is known for artificial intelligence and neural networks, the basis of blending psychology and codes together still contains a core aspect of [advanced statistics](https://www.sciencedirect.com/topics/social-sciences/neural-network). While most recognize the whole concept of neural networks is to mimic a (human) brain, most of TensorFlow and similar libraries functions are based on areas of math. Since you do not agree, instead of statistics I can change this term to artificial intelligence and machine learning library.

Reply:
Tensorflow is a symbolic math library based on dataflow and differentiable programming.